src/docs/binding-overview.md

```diff
@@ -0,0 +1,105 @@
+- Language parity mismatches
+- ABI (application binary interface) mismatches
+- Runtime differences
+- Idiomatic differences
+- Accessibility
+
+## Language Parity Mismatches
+Swift is a hybrid programming language leaning towards functional. It has all the features that you would expect in a traditional object oriented programming language, but from Apple's point of view, these are not features that are encouraged. Instead, Apple is encouraging using value types for most everything using copy-on-write semantics for the collection types. In fact, the main runtime for swift, swiftCore, contains 0 published classes as of a few years ago. In addition, Apple encourages "aspect oriented" rather than object oriented coding.
+
+In addition, Swift has a number of language features that do not exist in C#:
+- Discriminated Unions - Swift enums are built so that any case in the set may have payloads and the payloads need not be the same type
+- Actors - Swift has a flavor of heap-allocated types called Actors. There are largely syntactic sugar onto a non-virtual class, but there are
+restrictions generated by the compiler to effectively give an implicit lock.
+- Protocols with Associated Types (PATs) - these are a way of expressing recursive relations in generic interfaces in a less complicated way. One thing to keep in mind with PATs is that they are not actual types but rather a set of traits that types can adopt.
+- Protocol list types - this is a similar to having multiple interface constraints on a generic.
+- `some` - this is a two-purpose keyword. When used to modify an argument type this is syntactic sugar for an implicitly generic argument with a protocol constraint. When used to modify a return type, it represents a generic type whose provenance is owned by the caller instead of the callee
+- In calling a function that throws, the callee must either handle the error or declare that it rethrows
+- Naming - Swift allows all manner of unicode characters in function and operator names and the results are not pronounceable in C#
+- Homonyms - Swift allows functions to have identical names and arguments, but differ only in return type
+- Operators - Swift allows much more flexibility in defining operators than C#
+- Failable constructors - Swift allows constructors to be defined as `init?()` which will return `nil` on failure
+- Swift has only 1 namespace, the module name, with no nesting.
+- Swift has top level entities, i.e., global functions and variables.
+- In Swift, all members in a type *must* be initialized. Similarly, unsafe pointer types have a notion of 4 possible states that they can be in:
+  - allocated
+  - initialized
+  - deinitialized
+  - deallocated
+Effectively, all types follow this pattern, but the 4 states are managed by the compiler and the runtime and unlike pointer types, there is no explicit control of it.
+
+## ABI Differences
```

Comment:
I would like to see some primary document references. There are a lot of intricate details described here and understanding their provenance will help us move forward with addressing/accepting them as Swift interop evolves.